pkg/ingestor/collector/collector.go

```diff
@@ -15,4 +15,40 @@
 
 package collector
 
-type Collector interface{}
+import (
+	"context"
+
+	"github.com/guacsec/guac/pkg/ingestor/processor"
+)
+
+const (
+	BufferChannelSize int = 1000
+)
+
+type Collector interface {
+	// Retrieve the artifacts from the collector
```

Comment:
Add a bit more detail about how this should block until everything is done and return nil or error in either case. And also talk about how documents should be emitted to the channel.

Also golang style comments `// RetrieveArtifacts ...` so it gets picked up by godoc!

Reply:
added comment.